src/gt4py/next/common.py

```diff
@@ -189,11 +189,12 @@ def __and__(self, other: UnitRange) -> UnitRange:
         return UnitRange(max(self.start, other.start), min(self.stop, other.stop))
 
     def __contains__(self, value: Any) -> bool:
-        return (
-            isinstance(value, core_defs.INTEGRAL_TYPES)
-            and value >= self.start
-            and value < self.stop
-        )
+        # TODO(egparedes): use core_defs.IntegralScalar for `isinstance()` checks (see PEP 604)
+        #   and remove int cast, once the related mypy bug (#16358) gets fixed
+        if isinstance(value, core_defs.INTEGRAL_TYPES):
+            return self.start <= cast(int, value) < self.stop
+        else:
+            return False
```

Comment:
is this change from a merge conflict or is it intended?

Reply:
It is intended. Meant to address this error:
```
src/gt4py/next/common.py:194:17: error: Unsupported operand types for <= ("int" and "object")  [operator]
src/gt4py/next/common.py:194:17: error: Unsupported operand types for <= ("Infinity" and "object")  [operator]
src/gt4py/next/common.py:195:17: error: Unsupported operand types for > ("int" and "object")  [operator]
src/gt4py/next/common.py:195:17: error: Unsupported operand types for > ("Infinity" and "object")  [operator]
```

Reply:
Ok, now I got it (was confused by the `TODO(egparedes)`)